firecrown/likelihood/gauss_family/gauss_family.py

```diff
@@ -42,35 +43,85 @@ def read(self, sacc_data: sacc.Sacc) -> None:
         """Read the covariance matrix for this likelihood from the SACC file."""
 
         _sd = sacc_data.copy()
-        inds_list = []
+        indices_list = []
         for stat in self.statistics:
             stat.read(sacc_data)
-            inds_list.append(stat.sacc_inds.copy())
+            indices_list.append(stat.sacc_indices.copy())
 
-        inds = np.concatenate(inds_list, axis=0)
-        cov = np.zeros((len(inds), len(inds)))
-        for new_i, old_i in enumerate(inds):
-            for new_j, old_j in enumerate(inds):
+        indices = np.concatenate(indices_list, axis=0)
+        cov = np.zeros((len(indices), len(indices)))
+        for new_i, old_i in enumerate(indices):
+            for new_j, old_j in enumerate(indices):
                 cov[new_i, new_j] = _sd.covariance.dense[old_i, old_j]
         self.cov = cov
         self.cholesky = scipy.linalg.cholesky(self.cov, lower=True)
         self.inv_cov = np.linalg.inv(cov)
 
     @final
-    def compute_chisq(self, cosmo: pyccl.Cosmology) -> float:
+    def get_cov(self) -> np.ndarray:
+        """Gets the current covariance matrix."""
+        assert self.cov is not None
+        return self.cov
+
+    @final
+    def get_data_vector(self) -> np.ndarray:
+        """Get the data vector from all statistics
+        and concatenate in the right order."""
+        data_vector_list: List[np.ndarray] = []
+        for stat in self.statistics:
+            data = stat.get_data_vector()
+            data_vector_list.append(np.atleast_1d(data))
+        return np.concatenate(data_vector_list)
+
+    @final
+    def compute_theory_vector(self, cosmo: pyccl.Cosmology) -> np.ndarray:
+        """Computes the theory vector using the current instance of pyccl.Cosmology.
+
+        :param cosmo: CCL cosmology object to be used to compute the likelihood
+        """
+        theory_vector_list: List[np.ndarray] = []
+        for stat in self.statistics:
+            theory = stat.compute_theory_vector(cosmo)
+            theory_vector_list.append(np.atleast_1d(theory))
+
+        return np.concatenate(theory_vector_list)
+
+    @final
+    def compute(self, cosmo: pyccl.Cosmology) -> Tuple[np.ndarray, np.ndarray]:
         """Calculate and return the chi-squared for the given cosmology."""
```

Comment:
The comment needs updating to reflect the new purpose of compute.

Reply:
Done!